tracer/src/Datadog.Trace/RuntimeMetrics/RuntimeMetricsWriter.cs

```diff
@@ -129,14 +130,31 @@ internal RuntimeMetricsWriter(IDogStatsd statsd, TimeSpan delay, bool inAzureApp
 
         public void Dispose()
         {
-            AppDomain.CurrentDomain.FirstChanceException -= FirstChanceException;
+            if (Interlocked.CompareExchange(ref _disposed, 1, 0) == 1)
+            {
+                Log.Debug("Disposing Runtime Metrics but it was already disposed");
+                return;
+            }
+
+            Log.Debug("Disposing Runtime Metrics");
             _timer.Dispose();
+            AppDomain.CurrentDomain.FirstChanceException -= FirstChanceException;
+            // We don't dispose runtime metrics on .NET Core because of https://github.com/dotnet/runtime/issues/103480
+#if NETFRAMEWORK
             _listener?.Dispose();
+#endif
             _exceptionCounts.Clear();
+            _statsd.Dispose();
```

Comment:
I'm not sure that this is safe to call unfortunately 😟 The `IDogStatsd` instance is used in _multiple_ components, and disposing it here could cause failures in other components... it is probably better to move responsibility for disposing the `IDogStatsd` instance _out_ of `RuntimeMetricsWriter` entirely... That way you can ensure all the dependencies have been disposed _before_ disposing the `IDogStatsd` instance.